AWS_CloudWatch/Functions/Metrics.py

```diff
@@ -0,0 +1,24 @@
+from Functions.Metric import Metric
+
+class CounterMetric(Metric):
+
+    def __init__(self, name_space: str):
+        super().__init__(name_space, "Count")
+        self.counter = 0
+
+    def publish_metric(self, metric_name: str):
```

Comment:
this seems somewhat confusing - the publishing logic should be in the publisher (or alternatively, the Metric constructor should take a Publisher and use it to automatically publish)

with this we are not here nor there